### ISSUE TITLE
New Release?

### ISSUE THREAD

**Issue by skevy** [facebook/watchman#665]
Hey there! Thanks as always for the great work on Watchman.

It seems that v4.9 was released well over a year ago (in Aug of 2017). There's been a ton of work here since -- any chance we could get a new release?

Thanks!
-Adam

**Comment by wez** [comment]
Yeah, we're overdue on making a release.  I've been shepherding some changes through our internal CI to flip our build over to cmake and depend on folly and thrift in order to add support for our Eden virtual filesystem.  I'd like to get those landed and the CI happy and then I think we'll be in a good spot to review and make a release in the next couple of weeks.

**Comment by skevy** [comment]
Thanks so much for the update @wez! I'll check back in a few weeks :)

**Comment by alexandertrefz** [comment]
@wez How is the situation looking? Is there a release in sight, now that a few weeks have passed? Maybe even just a beta?

**Comment by wez** [comment]
There are a couple of things to get squared away before we can make a new release:

- The new windows build now needs to ship some dll deps.  You'll see a couple of duplicate issues that have been opened about missing dlls.  That needs to be run down and sorted out; in theory those should now be in the zip file, but appveyor hasn't had a successful test run since that change went in (it builds and the tests pass, but somehow the test run doesn't terminate), so the zip file linked from the download page are currently broken.
- The homebrew watchman recipe needs to be updated to use the new cmake build process and pull in at least the folly dep

We have higher internal priorities than these at the moment, so we don't have anyone actively working on this, but the good news is that both of these are potentially doable by anyone in the open source community, so if you have some bandwidth to try and help move us closer, I'll happily review PRs or answer questions that help someone put together a PR.

Please note that internally at FB we deploy `master` to a large number of machines every week (and have done for several years).  What is on the `master` branch is considered to be production ready for Linux and Mac, so you should generally be fine to roll a build off master and run it even though we haven't labelled it with its own version number.  (We don't use version numbers internally: we use the date of the build).

**Comment by skevy** [comment]
Hey @wez - 

Tried to do a build from source (on mac) today. Had a bunch of trouble and wasn't able to get past it.

First error was the error described here: https://github.com/facebook/folly/issues/652

Then I fuddled with getdeps.py a bit, and got past this folly error, but then I got errors saying: `ld: library not found for -lgflags_shared` when compiling other deps.

Is this something you could look into sometime soon?

**Comment by MiroFurtado** [comment]
Have also experienced the issue with jemalloc (on Mac)

**Comment by sluongng** [comment]
Bumping this as I have seen a lot of interesting commit that link Watchman with mononoke and eden. It would be nice to have a release build since the 2017 version?

**Comment by zoidbergwill** [comment]
@skevy Any chance you could push your attempts so I could have a go from there instead of from master, if you made some progress?

**Comment by sluongng** [comment]
Looks like the new release SOP is by publishing artifacts from master branch "CI=passed" github action

I.e.: https://github.com/facebook/watchman/actions/runs/31901438 -> Artifacts

You can obtain the artifact programmatically with the latest github api released yesterday (27th Jan): https://developer.github.com/v3/actions/artifacts/

**Comment by shepting** [comment]
It looks like you can actually click that artifact and download it.

**Comment by bric3** [comment]
Especially since this is hard to get the 4.9.0 binary. Or event to get a working lib following the [current documentation(https://facebook.github.io/watchman/docs/install#binary-downloads-for-linux-macos-and-windows-beta). Not all build are successful, and even those that are may not have a valid linux artifact. Or the artifact has been expired (e.g. the [last CI job at this time](https://github.com/facebook/watchman/runs/412005697?check_suite_focus=true) to have a successful build).

**Comment by wez** [comment]
Starting today, we have weekly tags with binaries uploaded to them: https://github.com/facebook/watchman/releases/latest

We deploy watchman continuously at FB and the weekly tags are an approximation of that outside FB.   A given weekly tag may be missing a binary for some platforms until we smooth out the open source build automation (eg: the macOS build didn't succeed this week), so the existence of a tag doesn't necessarily imply that everything is 100%.

Docs will be updated later in the week.

**Comment by bric3** [comment]
That's good news, thanks!
